docs/DeveloperGuide.md

```diff
@@ -695,3 +695,77 @@ testers are expected to do more *exploratory* testing.
    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases …​ }_
+
+### Mass Operations [Proposed]
+
+#### What is the feature about
+Provides a way for users to mark and unmark multiple `Task` at a time. The current implementation of mark and unmark
+users have to mark and unmark multiple tasks as completed or uncompleted individually, one at a time.
+
+#### How the feature is implemented
+
+The feature is to be implemented with the addition of a `MassOpsParser` class which parses through user inputs
+consisting of multiple indexes and processes the indexes to return an `ArrayList` of `Indexes` for `MarkCommand` and
+`UnmarkCommand` to execute on.
+
+##### MassOps: Marking tasks
+Given below is an example usage scenario of how the MassOps mechanism behaves at each step to mark tasks in
+the task list.
+
+Step 1. User inputs `mark 1 2 3` to mark tasks 1, 2 and 3 of the task list.
+
+Step 2. Upon receiving the user's input, `LogicManager` calls `HarmoniaParser#parseCommand()` to parse the user input.
+
+Step 3. The first word of the user input is `mark`, which matches the command for `MarkCommand`.
+
+Step 4. `MarkCommandParser#parse()` is called and `MassOpsParser#massOpsProcessor` is invoked to process the user input
+into `indexes`, an `ArrayList<Index>` containing the indexes to be marked.
+
+Step 5. `MarkCommand` is invoked upon `indexes` and returned to LogicManager.
+
+Step 6. After `MarkCommand#execute()` is executed, Harmonia retrieves each task to be marked from `lastShowList` and
+`createMarkedTask` is called to mark each task respectively.
+
+Step 7. Each command result is stored in `markedTasks` which is returned and displayed to the user at the
+end of the execution.
+
+![MassOpsMark]
+(images/MassOpsMark.png)
+
+##### MassOps: Unmarking tasks
+Given below is an example usage scenario of how the MassOps mechanism behaves at each step to unmark tasks in
+the task list.
+
+Step 1. User inputs `unmark 1 2 3` to mark tasks 1, 2 and 3 of the task list.
```

Comment:
Should `mark` be `unmark`, since the command is `unmark 1 2 3`?

> Step 1. User inputs `unmark 1 2 3` to **unmark** tasks 1, 2 and 3 of the task list.